### PR TITLE
Jenkins 21750 - Support Promoted Builds Plugin

### PR DESCRIPTION
**Pull request by apottere**
I took a stab at updating the dsl plugin to allow creation of promotions from the [Promoted Builds Plugin](https://wiki.jenkins-ci.org/display/JENKINS/Promoted+Builds+Plugin), since that issue has been open for a while now ([JENKINS-21750](https://issues.jenkins-ci.org/browse/JENKINS-21750)).

With this method, promotion steps can be declared as follows:

```
job {
    name 'My Cool Job That Gets Promoted'
   
    promotion('My Promotion's Name') {
        configure { root ->
            root / 'buildSteps' / 'hudson.tasks.Shell' {
                command 'echo "hello, world!"'
            }
        }
    }
}
```

Currently, only the configure block is supported.  While this isn't ideal, it's a good starting point and it allows fully functional promotion processes with a little bit of work.  Going forward, more convenience methods can be added to the PromotionContext to make it easier to work with.

Provided are two tests showing the Promotion object can create an XML config from a template much like any of the Jobs.

### CODE REVIEW

**Comment by pascalmartin** [comment]
I test the code i my fork but i need to made "Reload Configuration from Disk" in jenkins admin menu after de dsl seeed to make promotion working. Did I forget something?
https://github.com/pascalmartin/job-dsl-plugin

Is this feature will be integrated in the next release?

**Comment by pascalmartin** [comment]
The new Promotion-template.xml have two time <properties/> an cause error

File: job-dsl-core/src/main/resources/javaposse/jobdsl/dsl/jobs/Promotion-template.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    <scm class="hudson.scm.NullSCM"/>
+    <canRoam>false</canRoam>
+    <disabled>false</disabled>
+    <blockBuildWhenDownstreamBuilding>false</blockBuildWhenDownstreamBuilding>
+    <blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding>
+    <triggers/>
+    <conditions/>
+    <icon/>
+    <buildSteps/>
+    <properties/>
```
Comment (pascalmartin) [inline job-dsl-core/src/main/resources/javaposse/jobdsl/dsl/jobs/Promotion-template.xml]: Duplicate properties node ? Merge error ?

**Comment by apottere** [comment]
That error was most likely just poor proofreading.

Shortly after creating this patch, I found a thread where the maintainers of the job dsl plugin said they were going to forego adding explicit job dsl support for promotions in favor of plugin support.  That means this will most likely never get pulled - which is fine.

This is simply a stopgap until that functionality exists for real.

**Comment by daspilker** [comment]
@denschu is working on implementing a DSL extension point for the Promoted Builds Plugin, see [JENKINS-21750](https://issues.jenkins-ci.org/browse/JENKINS-21750?focusedCommentId=234593&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-234593).

**Pull request closed without merging**